client/src/components/Ticketing/ticketingmanager/TicketTypes/SubscriptionTypes.tsx

```diff
@@ -0,0 +1,374 @@
+import React, {useState} from 'react';
+import PopUp from '../../PopUp';
+import {
+  DataGrid,
+  GridColumns,
+  GridEditInputCell,
+  GridEventListener,
+  GridPreProcessEditCellProps,
+  GridRenderEditCellParams,
+  GridRowEditStopReasons,
+  GridRowId,
+  GridRowModel,
+  GridRowModes,
+  GridRowModesModel,
+  GridValueFormatterParams,
+  useGridApiContext,
+} from '@mui/x-data-grid';
+import {useFetchToken} from '../Event/components/ShowingUtils';
+import {Switch, Tooltip} from '@mui/material';
+import {FormButton} from '../Event/components/FormButton';
+import {FormDeleteButton} from '../Event/components/FormDeleteButton';
+import {formatUSD} from '../RefundOrders/RefundOrders';
+import {
+  apiCall,
+  getDefaultSubscriptionType,
+  useFetchSubscriptionTypes,
+  usePopUp,
+  validateSubscriptionType,
+} from './SubscriptionTypeUtils';
+import {
+  BackIcon,
+  CirclePlusIcon,
+  EditIcon,
+  SaveIcon,
+  TrashCanIcon,
+} from '../../Icons';
+
+export const SubscriptionTypes = () => {
+  const {token} = useFetchToken();
+  const {popUpProps, setPopUpProps, showPopUp, setShowPopUp} = usePopUp();
+  const {subscriptionTypes = [], setSubscriptionTypes} =
+    useFetchSubscriptionTypes();
+  const [rowModesModel, setRowModesModel] = useState<GridRowModesModel>({});
+  const [editing, setEditing] = useState(false);
+
+  const handleRowModesModelChange = (newRowModesModel: GridRowModesModel) => {
+    setRowModesModel(newRowModesModel);
+  };
+
+  const handleRowEditStart: GridEventListener<'rowEditStart'> = (_, event) => {
+    if (editing) {
+      event.defaultMuiPrevented = true;
+    } else {
+      setEditing(true);
+    }
+  };
+
+  const handleDeleteClick = (id: GridRowId) => {
+    const {id: current, ...rest} = rowModesModel;
+    setPopUpProps(
+      'Confirm Delete',
+      `Press continue to proceed with deletion`,
+      false,
+      `delete-subscription-row-${id}`,
+      async () => {
+        try {
+          await apiCall(
+            'DELETE',
+            `${process.env.REACT_APP_API_2_URL}/subscription-types/${id}`,
+            undefined,
+            token,
+          );
+          setRowModesModel(rest);
+          setSubscriptionTypes((types) =>
+            types.filter((type) => type.id !== id),
+          );
+        } catch (error) {
+          setPopUpProps(
+            'Delete Failed',
+            error.json
+              ? (await error.json()).error
+              : 'Error deleting subscription type',
+            false,
+            'delete-failed',
+          );
```

Comment:
On an informational only popup, we usually only want to have the primary button, and leave out the secondary button and close button.

Reply:
Fixed